encoding/thrift/options.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package thrift
+
+type clientConfig struct {
+	// TODO: disableEnveloping bool
+}
+
+// ClientOption customizes the behavior of a Thrift client.
+type ClientOption struct {
+	apply func(*clientConfig)
+}
+
+type registerConfig struct {
+	// TODO: disableEnveloping bool
+}
+
+// RegisterOption customizes the behavior of a Thrift handler during
+// registration.
+type RegisterOption struct {
+	apply func(*registerConfig)
```

Comment:
why a struct instead of an interface with a private function? With an interface, we could have a single `Option` interface which embeds both `RegisterOption` and `ClientOption` (`apply` would be a different function name in both), and have something like `DisableEnveloping` be an `Option` that can be used for both.